drivers/adc/adc_stm32.c

```diff
@@ -997,17 +971,17 @@ static int start_read(const struct device *dev,
 
 #if defined(CONFIG_SOC_SERIES_STM32G0X) || \
 	defined(CONFIG_SOC_SERIES_STM32WLX)
-	if (LL_ADC_GetResolution(adc) != resolution) {
+	if (get_resolution(dev) != resolution) {
 		/*
 		 * Writing ADC_CFGR1 register while ADEN bit is set
 		 * resets RES[1:0] bitfield. We need to disable and enable adc.
 		 */
 		adc_stm32_disable(adc);
-		LL_ADC_SetResolution(adc, resolution);
+		set_resolution(dev, resolution);
 	}
 #elif !defined(CONFIG_SOC_SERIES_STM32F1X) && \
 	!defined(STM32F3X_ADC_V2_5)
-	LL_ADC_SetResolution(adc, resolution);
+	set_resolution(dev, resolution);
 #endif
```

Comment:
I haven't touched the logic here, but I hesitate to remove all #if, add a simple if to check whether config->res_reg is equal to 0xFF (to filter out STM32F1 and F334), and to apply the same logic of read, disable, set for all other series.
Any thoughts?